src/Modules/UngradedModules/ModuleGens.jl

```diff
@@ -401,31 +401,74 @@ Note: `:via_lift` is typically faster than `:via_transform` for a single vector
 is faster if many vectors are lifted
 """
 function coordinates(a::FreeModElem, M::SubModuleOfFreeModule, task::Symbol = :auto)
-  if iszero(a)
-    return sparse_row(base_ring(parent(a)))
-  end
-  if task == :auto
-    if coefficient_ring(base_ring(parent(a))) isa Field #base_ring(base_ring(...)) does not work for MPolyQuos
-      task = :via_transform
+    R = base_ring(parent(a))
+    if iszero(a)
+        return sparse_row(R)
+    end
+    for (i, g) in enumerate(gens(M))
+        if a == g
+            row = sparse_row(R)
+            push!(row.pos, i)
+            push!(row.values, one(R))
+            return row
+        end
+    end
+    return coordinates_atomic(a, M; task=task)
+end
+
+
+function coordinates_atomic(a::FreeModElem{T}, M::SubModuleOfFreeModule; task::Symbol = :auto) where {S<:Union{ZZRingElem,<:FieldElem}, T<:MPolyRingElem{S}}
```

Comment:
```suggestion
function coordinates_atomic(a::FreeModElem{T}, M::SubModuleOfFreeModule; task::Symbol = :auto) where {S<:Union{ZZRingElem,FieldElem}, T<:MPolyRingElem{S}}
```
A union does not need the `<:` inside, even worse, it sometimes confuses the compiler. So better to avoid it in places like this (not only here, many more places in this pr)